### ISSUE TITLE
Error content-length header

### ISSUE THREAD

**Issue by r-zlotorzynski** [fhessel/esp32_https_server#45]
** My test environment  **
ESP32  WROVER 16

When trying to set a Content-length header such as 1, 10, 1000, 10000 ... the header is incorrectly formatted.

This function is ./src/util.cpp    (intToString)

```
std::string intToString(int i) {
  if (i==0) {
    return "0";
  }
  // We need this much digits
  int digits = ceil(log10(i));
  char c[digits+1];
  c[digits] = '\0';

  for(int x = digits-1; x >= 0; x--) {
    char v = (i%10);
    c[x] = '0' + v;
    i = (i-v)/10;
  }

  return std::string(c);
}
```

The results of this function are exemplary data:

![BEFORE](https://user-images.githubusercontent.com/26817418/61174864-7950bf80-a5a6-11e9-914b-e0eaff13c3cf.PNG)

For the above-mentioned numbers, the values are incorrect. In addition, no value is returned for length 1.

I have slightly changed the function on:
```
/**
    * TODO: Convert int to string C++
    *
    * @param            {int}                           i                              Int >= 0
    * @return             {std::string}
    * @public
*/
std::string intToString( int i ) {
    // Value is 0, return const 0
    if ( i <= 0 ) return std::string( "0" );

    // Prepare buffer
    char buffer[ 8 ];         

    // Copy INT to buffer
    itoa( i , buffer , 10 );

    // Return INT as C++ string
    return std::string( buffer );
}
```

After the change, the results look better

![AFTER](https://user-images.githubusercontent.com/26817418/61174994-53c4b580-a5a8-11e9-8203-458a77956cd5.PNG)

CODE TESTING:
```
    Serial.println( "Start calc:  0" );
    std::string _temp = int_to_cpp_string( 0 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  1" );
    _temp = int_to_cpp_string( 1 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  2" );
    _temp = int_to_cpp_string( 2 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }


    Serial.println( "Start calc:  10" );
    _temp = int_to_cpp_string( 10 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  11" );
    _temp = int_to_cpp_string( 11 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  1000" );
    _temp = int_to_cpp_string( 1000 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  1001" );
    _temp = int_to_cpp_string( 1001 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  10000" );
    _temp = int_to_cpp_string( 10000 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }

    Serial.println( "Start calc:  10001" );
    _temp = int_to_cpp_string( 10001 );
    for (int i = 0; i < _temp.length(); i++)
    {
        Serial.println(_temp[i]);
    }
```

**Comment by fhessel** [comment]
Thanks for finding this and the detailed information, it seems to be a typical off-by-one error.

The reason why I did not use `itoa` is the missing check for array bounds. Your example code will have problems with value bigger than 9.999.999 (even though that might be a theoretical problem only on the ESP32). You'd have to use the right length there too, which will make you do the same computation with `log10` that I did (or go for the maximum for type int, but I don't like to include architecture-dependent fixed numbers).

I'll have a look at it and fix the bug, but I don't have hardware for testing it at hand right now.
